hydra-node/test/Hydra/API/ServerSpec.hs

```diff
@@ -95,25 +102,83 @@ spec = parallel $ do
       monitor $ cover 0.1 (null outputs) "no message when reconnecting"
       monitor $ cover 0.1 (length outputs == 1) "only one message when reconnecting"
       monitor $ cover 1 (length outputs > 1) "more than one message when reconnecting"
-      run . failAfter 5 $ do
+      run . failAfter 15 $ do
         withFreePort $ \port ->
           withAPIServer @SimpleTx "127.0.0.1" (fromIntegral port) alice mockPersistence nullTracer noop $ \Server{sendOutput} -> do
             mapM_ sendOutput outputs
-            withClient port $ \conn -> do
+            withClient port defaultPath $ \conn -> do
               received <- replicateM (length outputs + 1) (receiveData conn)
               case traverse Aeson.eitherDecode received of
                 Left{} -> failure $ "Failed to decode messages:\n" <> show received
-                Right timedOutputs ->
+                Right timedOutputs -> do
                   (output <$> timedOutputs) `shouldBe` greeting : outputs
 
+  it "does not echo history if client says no" $
+    checkCoverage . monadicIO $ do
+      outputs :: [ServerOutput SimpleTx] <- pick arbitrary
+      monitor $ cover 0.1 (null outputs) "no message when reconnecting"
+      monitor $ cover 0.1 (length outputs == 1) "only one message when reconnecting"
+      monitor $ cover 1 (length outputs > 1) "more than one message when reconnecting"
+      run . failAfter 5 $ do
+        withFreePort $ \port ->
+          withAPIServer @SimpleTx "127.0.0.1" (fromIntegral port) alice mockPersistence nullTracer noop $ \Server{sendOutput} -> do
+            -- send arbitrary outputs from the server (which should be ignored by the client)
+            mapM_ sendOutput outputs
+            -- start client that doesn't want to see the history
+            withClient port (defaultPath <> "?history=0") $ \conn -> do
+              -- receive only one message which should be the greeting one
+              receivedGreeting <- replicateM 1 (receiveData conn)
+
+              case traverse Aeson.eitherDecode receivedGreeting of
+                Left{} -> failure $ "Failed to decode messages:\n" <> show receivedGreeting
+                Right timedOutputs -> do
+                  (output <$> timedOutputs) `shouldBe` [greeting]
+
+              -- send another message from server
+              newOutput :: ServerOutput SimpleTx <- generate arbitrary
+              sendOutput newOutput
+
+              -- Receive one more message and expect it to be the same as sent one.
+              -- This means other messages we sent after the 'Greeting' are ignored as expected.
```

Comment:
I find this comment confusing as once connected you will start receiving outputs.
I would paraphrase it.